lale/lib/lale/tee.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2019 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import lale.docstrings
+import lale.operators
+
+
+class _TeeImpl:
+    def __init__(self, observer=None):
+        self._observer = observer
+
+    def transform(self, X, y=None):
+        if self._observer is not None:
+            self._observer(X, y)
+        return X
+
+    def transform_schema(self, s_X):
+        """Used internally by Lale for type-checking downstream operators."""
+        return s_X
+
+
+_hyperparams_schema = {
+    "allOf": [
+        {
+            "description": "This first sub-object lists all constructor arguments with their "
+            "types, one at a time, omitting cross-argument constraints, if any.",
+            "type": "object",
+            "additionalProperties": False,
+            "relevantToOptimizer": [],
+            "properties": {
+                "observer": {"anyOf": [{"laleType": "callable"}, {"enum": [None]}]}
```

Comment:
Missing "description" field. In fact, I would put two: one for the callable case, and a different one for the None case.

Reply:
added